framework/source/class/qx/ui/table/model/Filtered.js

```diff
@@ -99,18 +109,11 @@ qx.Class.define("qx.ui.table.model.Filtered",
      * @param the_needle {String} String to search
      * @param the_haystack {Array} Array, which should be searched
      * @return {Boolean} whether the search string was found.
+     * @deprecated {6.0} Use qx.lang.Array.contains instead
      */
     _js_in_array : function(the_needle, the_haystack)
     {
-      var the_hay = the_haystack.toString();
-
-      if (the_hay == '') {
-        return false;
-      }
-
-      var the_pattern = new RegExp(the_needle, 'g');
-      var matched = the_pattern.test(the_haystack);
-      return matched;
+      return qx.lang.Array.contains(the_haystack, the_needle);
```

Comment:
On Wed, 6 Jul 2016, William Oprandi wrote:

>>       */
>>      _js_in_array : function(the_needle, the_haystack)
>>      {
>> -      var the_hay = the_haystack.toString();
>> -
>> -      if (the_hay == '') {
>> -        return false;
>> -      }
>> -
>> -      var the_pattern = new RegExp(the_needle, 'g');
>> -      var matched = the_pattern.test(the_haystack);
>> -      return matched;
>> +      return qx.lang.Array.contains(the_haystack, the_needle);
>
> The behavior is the same :
> `new RegExp("a", "g").test(["a", "b", "c"]) === true`
> `new RegExp("c", "g").test(["a", "b", "c"]) === true`
> `new RegExp("d", "g").test(["a", "b", "c"]) === false`

How about

_js_in_array("ab", ["ab",  "c"]);

and

_js_in_array("ab", ["a", "b", "c"]);

Are those suppossed to give the same results?


-- 
Oetiker+Partner AG              tel: +41 62 775 9903 (direct)
Fritz Zaucker                        +41 62 775 9900 (switch board)
Aarweg 15                            +41 79 675 0630 (mobile)
CH-4600 Olten                   fax: +41 62 775 9905
Schweiz                         web: www.oetiker.ch